cid/helpers/quicksight/__init__.py

```diff
@@ -199,14 +199,21 @@ def discover_dashboard(self, dashboardId: str) -> Dashboard:
             raise CidCritical(f'Dashboard {dashboardId} was not found')
         try:
             _template_arn = dashboard.version.get('SourceEntityArn')
-            _template = self.describe_template(
-                template_id=_template_arn.split('/')[1],
-                account_id=_template_arn.split(':')[4],
-                region=_template_arn.split(':')[3],
-                version_number=int(_template_arn.split('/')[-1]),
-            )
-            if isinstance(_template, CidQsTemplate):
-                dashboard.deployedTemplate = _template
+            
+            ## Check for version_number
+            template_version_nb = int(_template_arn.split('/')[-1]) if _template_arn.split('/')[-1].isdecimal() else None
+            
+            if  template_version_nb:
```

Comment:
why not leaving template_version_nb as None?

describe works as expected

Reply:
If template_version_nb is None we still need to describe_template

Reply:
We discussed this and it looks like if version is none we get only the latest that does not make sense for further actions.